pkg/service/FluxService/FluxApplicationService.go

```diff
@@ -0,0 +1,420 @@
+	if err != nil {
+		impl.logger.Errorw("error in getting resource", "err", err)
+		return nil, nil, err
+	}
+
+	releaseName := ""
+	_namespace := ""
+	var appStatus *FluxAppStatusDetail
+
+	if resp != nil && resp.Manifest.Object != nil {
+		releaseName, _namespace, err = getReleaseNameNamespace(resp.Manifest.Object, request.Name)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		appStatus, err = getKsAppStatus(resp.Manifest.Object, GvkForHelmreleaseFluxApp)
+		if err != nil {
+			impl.logger.Errorw("error in getting app status", "err", err)
+			return nil, nil, err
+		}
+
+		if releaseName == "" && _namespace == "" {
+			impl.logger.Errorw("error in getting release name and namespace", "err", err)
+			return nil, appStatus, err
+		}
+	}
+	helmRelease, err := impl.common.GetHelmRelease(request.Config, _namespace, releaseName)
+	if err != nil {
+		impl.logger.Errorw("error in getting helm release", "err", err)
+		return nil, appStatus, err
+	}
+	appDetailRequest := &client.AppDetailRequest{
+		ReleaseName:   releaseName,
+		Namespace:     _namespace,
+		ClusterConfig: request.Config,
+	}
+
+	resourceTreeResponse, err := impl.common.BuildResourceTree(appDetailRequest, helmRelease)
+	if err != nil {
+		impl.logger.Errorw("error in building resource tree", "err", err)
+		return nil, appStatus, err
+	}
+	fluxAppTreeResponse = append(fluxAppTreeResponse, resourceTreeResponse)
+
+	return fluxAppTreeResponse, appStatus, nil
+}
+func (impl *FluxApplicationServiceImpl) buildFluxAppDetailForKustomize(request *FluxAppDetailRequest) ([]*bean.ResourceTreeResponse, *FluxAppStatusDetail, error) {
+	fluxAppTreeResponse := make([]*bean.ResourceTreeResponse, 0)
+
+	k8sClusterConfig := impl.converter.GetClusterConfigFromClientBean(request.Config)
+	restConfig, err := impl.k8sUtil.GetRestConfigByCluster(k8sClusterConfig)
+	if err != nil {
+		impl.logger.Errorw("error in getting the rest config", "error", err)
+		return nil, nil, err
+	}
+	resp, err := impl.k8sUtil.GetResource(context.Background(), request.Namespace, request.Name, GvkForKustomizationFluxApp, restConfig)
+	if err != nil {
+		impl.logger.Errorw("error in getting app manifest", "err", err)
+		return nil, nil, err
+	}
+	if resp == nil || resp.Manifest.Object == nil {
+		impl.logger.Errorw("app manifest's response or manifest object is nil", "err", err)
+		return nil, nil, err
+	}
+	appStatus, err := getKsAppStatus(resp.Manifest.Object, GvkForKustomizationFluxApp)
+	if err != nil {
+		impl.logger.Errorw("failed to get app status", "err", err)
+		return nil, nil, err
+	}
+	if !inventoryExists(resp.Manifest.Object) {
+		impl.logger.Errorw("Inventory is empty ", err)
+		return nil, appStatus, err
+	}
+
+	fluxK8sResourceList := make([]*client.ExternalResourceDetail, 0)
+	fluxHrList := make([]*FluxHr, 0)
+	impl.getFluxResourceAndFluxHrList(request, &fluxK8sResourceList, &fluxHrList, resp.Manifest.Object)
```

Comment:
since fluxK8sResourceList and fluxHrList are slices not simple array, arrays in go are copied when passed via functions but slices behave differently in go, no need to pass slices by reference, it 's already passed by reference , hence any operation on these two slices inside that func will reflect the original array. 

also you simply could have done this :- 
fluxK8sResourceList, fluxHrList := impl.getFluxResourceAndFluxHrList(request, resp.Manifest.Object)

instead of what you have currently executed this would reduce code complexity, also this is a better approach rather than what you did